Sources/Support.swift

```diff
@@ -24,13 +26,14 @@ extension Dictionary {
 
 protocol ReadWriteLock {
     mutating func read<T>(_ block: () throws -> T) rethrows -> T
-    mutating func write(_ block: @escaping () -> Void, completion: (() -> Void)?)
+    mutating func write_async(_ block: @escaping () -> Void, completion: (() -> Void)?)
+    mutating func write_sync<T>(_ block: () throws -> T) rethrows -> T
```

Comment:
I'm thinking that it might be better to follow Dispatch here. The API for `read<T>` is the same as that for `write_sync<T>` so, how about we do this instead:

```swift
protocol ReadWriteLock {
    mutating func sync<T>(_ block: () throws -> T) rethrows -> T
    mutating func async(_ block: @escaping () -> Void, completion: (() -> Void)?)
}